File: Java/src/test/java/org/chornogora/task_16_3sum_closest/SolutionTest.java
package org.chornogora.task_16_3sum_closest;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;

public class SolutionTest {

    @Test
    public void shouldReturn_2() {
        Solution solution = new Solution();
        int result = solution.threeSumClosest(new int[]{-1, 2, 1, -4}, 1);
        Assertions.assertEquals(2, result);
    }

    @Test
    public void shouldReturn_0() {
        Solution solution = new Solution();
        int result = solution.threeSumClosest(new int[]{0, 0, 0}, 1);
        Assertions.assertEquals(0, result);
    }

    @Test
    public void shouldReturn_60() {
        Solution solution = new Solution();
        int result = solution.threeSumClosest(new int[]{10, 20, 30, 40, 50, 60, 70, 80, 90}, 1);
        Assertions.assertEquals(60, result);
    }
}

File: Kotlin/src/test/kotlin/org/chornogora/task_4_median_of_two_sorted_arrays/SolutionTest.kt
package org.chornogora.task_4_median_of_two_sorted_arrays

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class SolutionTest {

    val solution: Solution = Solution()

    @Test
    fun shouldReturnMiddleElementOfFirstArray() {
        val result = solution.findMedianSortedArrays(IntArray(3) { it }, intArrayOf())
        Assertions.assertEquals(1.0, result)
    }

    @Test
    fun shouldReturnMedianOfSecondArray() {
        val result = solution.findMedianSortedArrays(intArrayOf(), intArrayOf(4, 5, 6, 7))
        Assertions.assertEquals(5.5, result)
    }

    @Test
    fun shouldReturnMedianUsingOnlySecondArray() {
        val result = solution.findMedianSortedArrays(intArrayOf(1), intArrayOf(2, 5, 6, 7))
        Assertions.assertEquals(5.0, result)
    }

    @Test
    fun shouldReturnMedianUsingOnlyFirstArray() {
        val result = solution.findMedianSortedArrays(intArrayOf(1, 2, 3), intArrayOf(6, 7))
        Assertions.assertEquals(3.0, result)
    }

    @Test
    fun shouldReturnMedianUsingBothArrays() {
        val result = solution.findMedianSortedArrays(intArrayOf(1, 3, 4, 12, 18), intArrayOf(2, 6, 7, 8, 9, 16))
        Assertions.assertEquals(7.0, result)
    }

    @Test
    fun shouldReturnMedianUsingBothArraysOfTheSameSize() {
        val result = solution.findMedianSortedArrays(intArrayOf(1, 3, 4, 12, 18, 20), intArrayOf(2, 6, 7, 8, 9, 16))
        Assertions.assertEquals(7.5, result)
    }
}
